### ISSUE TITLE
dataProxy + ESbuild - PrismaClient is unable to be run in the browser

### ISSUE THREAD

**Issue by irvinebroque** [prisma/prisma#9894]
### Bug description

refs https://github.com/prisma/prisma/issues/8884, https://github.com/prisma/prisma/issues/85#issuecomment-947794166, https://github.com/prisma/prisma/pull/9561, https://github.com/prisma/prisma/issues/9841

Using Prisma + Cloudflare Workers + Planetscale. I'm using `dataProxy` in order to use Prisma within Cloudflare Workers. I keep hitting this when trying to build:

``` js
`Uncaught Error: PrismaClient is unable to be run in the browser.`
```

I'm generating the client with the `dataproxy` flag, and importing it normally:

``` 
PRISMA_CLIENT_ENGINE_TYPE=dataproxy npx prisma generate
```

``` javascript
import { PrismaClient } from "@prisma/client";
const prisma = new PrismaClient();
```

[These docs](https://www.prisma.io/docs/guides/deployment/deployment-guides/deploying-to-cloudflare-workers#11-develop-the-cloudflare-worker-function) on how to use Prisma + MongoDB + Cloudflare Workers are the closest to what I'm working with, but with mysql instead of mongodb.

### How to reproduce

1. Generate a Prisma Client using the `PRISMA_CLIENT_ENGINE_TYPE=dataproxy` flag
2. Import PrismaClient and initialize a new client instance (`import { PrismaClient } from "@prisma/client"; const prisma = new PrismaClient();`)
3. Attempt to bundle Worker code (ex: using ESbuild), and observe error:

```
💁  Uploading updated files...
Error: Something went wrong with the request to Cloudflare...
Uncaught Error: PrismaClient is unable to be run in the browser.
In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
  at line 15156 in PrismaClient2
  at line 27895
 [API code: 10021]
```

### Expected behavior

Expected that a Cloudflare Worker-compatible client would be generated, based on the changes introduced in https://github.com/prisma/prisma/pull/9561.

If the client was not properly generated, I'd have expected either a more specific error message, or to skip past the `PrismaClient is unable to be run in the browser` to allow me to debug what went wrong more easily. It seems possible that this generic error is hiding something more specific?

### Prisma information

From `prisma.schema`:

``` prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["referentialIntegrity", "dataProxy"]
  engineType = "dataproxy"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
  shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
  referentialIntegrity = "prisma"
}

```

Esbuild config, in case it's relevant:

``` js
esbuild
  .build({
    entryPoints: ["./src/worker/index.mjs"],
    format: "esm",
    bundle: true,
    outfile: "dist/index.mjs",
    inject: ["./src/worker/polyfills.mjs"],
    plugins: [graphqlLoaderPlugin.default()],
  })
```

### Environment & setup

- OS: Ubuntu-latest (Github Codespaces)
- Database: MySQL (Planetscale)
- Node.js version: 17.0.1

### Prisma Version

```
prisma                  : 3.3.0
@prisma/client          : 3.3.0
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 33838b0f78f1fe9052cf9a00e9761c9dc097a63c (at ../../usr/local/share/nvm/versions/node/v17.0.1/lib/node_modules/prisma/node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 33838b0f78f1fe9052cf9a00e9761c9dc097a63c (at ../../usr/local/share/nvm/versions/node/v17.0.1/lib/node_modules/prisma/node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine    : introspection-core 33838b0f78f1fe9052cf9a00e9761c9dc097a63c (at ../../usr/local/share/nvm/versions/node/v17.0.1/lib/node_modules/prisma/node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary           : prisma-fmt 33838b0f78f1fe9052cf9a00e9761c9dc097a63c (at ../../usr/local/share/nvm/versions/node/v17.0.1/lib/node_modules/prisma/node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Default Engines Hash    : 33838b0f78f1fe9052cf9a00e9761c9dc097a63c
Studio                  : 0.437.0
```

**Comment by irvinebroque** [comment]
Yep, here's what I get:

``` javascript
const prisma = new PrismaClient({
  errorFormat: "pretty",
  log: ["query", "info", "warn", "error"],
});

  try {
    const user = await prisma.user.findUnique({
      where: {
        id: 'example@prisma.io',
      },
    })
    console.log(user);
  } catch (e) {
    console.log(e.message, e.errorCode, e.code, e.meta);
  } finally {
    prisma.$disconnect();
  }
```

```
A hanging Promise was canceled. This happens when the worker runtime is waiting for a Promise from JavaScript to resolve, but has detected that the Promise cannot possibly ever resolve because all code and events related to the Promise's request context have already finished.
Error: The script will never generate a response. at line 0, col -2
{
  "exceptionDetails": {
    "columnNumber": -2,
    "exception": {
      "className": "Error",
      "description": "Error: The script will never generate a response.",
      "preview": {
        "description": "Error: The script will never generate a response.",
        "entries": null,
        "overflow": false,
        "properties": [
          {
            "name": "stack",
            "subtype": null,
            "type": "string",
            "value": "Error: The script will never generate a response.",
            "valuePreview": null
          },
          {
            "name": "message",
            "subtype": null,
            "type": "string",
            "value": "The script will never generate a response.",
            "valuePreview": null
          }
        ],
        "subtype": "error",
        "type": "object"
      },
      "subtype": "error",
      "type": "object",
      "value": null
    },
    "lineNumber": 0,
    "text": "Uncaught (in response)",
    "url": "undefined"
  },
  "timestamp": 1634950238157
}
```

Most of the comments on the Cloudflare forums like [this one](https://community.cloudflare.com/t/a-hanging-promise-was-canceled-rust-wasm-worker/282471/8) are WASM-related. I'll dig around and see what the issue might be.

**Comment by irvinebroque** [comment]
Resolved in version `3.4.1`. Excited to try this out!

**Comment by vinaypuppal** [comment]
I tried `3.4.1` with `esbuild` and it worked. I created a template starter project here https://github.com/vinaypuppal/cfw-prisma-template
